lib/v2/lofter/tag.js

```diff
@@ -1,59 +1,94 @@
 const got = require('@/utils/got');
 const cheerio = require('cheerio');
 const { parseDate } = require('@/utils/parse-date');
+const vm = require('node:vm');
 
 module.exports = async (ctx) => {
     const name = ctx.params.name ?? '摄影';
     const type = ctx.params.type ?? 'new';
+    const pageSize = 20;
+    const startingIndex = 0;
 
-    const rootUrl = 'https://www.lofter.com';
-    const currentUrl = `${rootUrl}/tag/${name}/${type}`;
+    const url = 'https://www.lofter.com/dwr/call/plaincall/TagBean.getCommonTagExcellentAuthors.dwr';
+
+    const body = [
+        'callCount=1',
+        'scriptSessionId=${scriptSessionId}187',
+        'httpSessionId=',
+        'c0-scriptName=TagBean',
+        'c0-methodName=search',
+        'c0-id=0',
+        `c0-param0=string:${name}`,
+        'c0-param1=number:0',
+        'c0-param2=string:',
+        `c0-param3=string:${type}`,
+        'c0-param4=boolean:false',
+        'c0-param5=number:0',
+        `c0-param6=number:${pageSize}`,
+        `c0-param7=number:${startingIndex}`,
+        'c0-param8=number:0',
+        'batchId=493053',
+    ];
 
     const response = await got({
-        method: 'get',
-        url: currentUrl,
+        method: 'post',
+        url: encodeURI(url),
+        body: body.join('\r'),
     });
 
-    const $ = cheerio.load(response.data);
-
-    $('.js-digest').remove();
+    const context = {
+        dwr: {
+            engine: {
+                _remoteHandleCallback() {
+                    this.data = arguments;
+                },
+            },
+        },
+    };
+    vm.createContext(context);
+    vm.runInContext(response.data, context);
+    const data = context.dwr.engine.data[2];
```

Comment:
Please use `jsdom` instead of `vm`.
```suggestion
    const dom = new JSDOM(
        `<script>if (dwr == null) var dwr = {};
        if (dwr.engine == null) dwr.engine = {};
        dwr.engine._remoteHandleCallback = function () {
            this.data = arguments;
        };
        ${response.data}</script>`,
        {
            runScripts: 'dangerously',
        }
    );
    const data = dom.window.dwr.engine.data[2];
```